### DISCUSSION TITLE
Proposal: a user-option to turn off drag bars

### DISCUSSION THREAD

**Discussion by petersampsonaudacity** [audacity/audacity#1492] [Development]
As a simple single-track user of Audacity I found myself recently  wanting to revert to 3.0.3 to do some simple editing as I didn't want or need the Drag-bar affordances.  

So I would like to propose an option to turn those off for a user like me who doesn't want or need them:

1) **the drag-bars would be "on" by default (for discoverability)**

2) a preference and/or a menu command to turn them off and on (Audacity "remembers" the setting)

3) right click in any Drag-bar for a dropdown menu with an entry for turning them off (cf. the Scrub Ruler which can be similarly turned off))

4) if a user creates a clip then the Drag-bars get turned on _(unless the Time Shift tool is retained for whatever reason)_

@Tantacrul and @jounih  what is the UX/GUI expert view on this?

**Comment by petersampsonaudacity** [comment]
**Afterthought:**  they may be less of a distraction when the Drag-bars are reduced to a single one for a stereo track and not one per channel - but I don't see any sign of that proposed idea appearing in master builds yet

**Comment by LWinterberg** [comment]
Assuming the time shift tool is getting removed in the medium term, how do
you plan on moving clips around in a single-track recording?

**Reply by petersampsonaudacity** [reply]
> Assuming the time shift tool is getting removed in the medium term, 
>how do you plan on moving clips around in a single-track recording?

"I" don't - I never use clips _(and I've only ever used timeshifting IRL once in all the 15 years I've been an Audacity user)_ - and I suspect many other users don't either.

But that's why I suggested "on" by default - so that those of us who know we never need them can turn them off, leaving them there for the more sophisticated users.

Plus I can move things around on a single track with Cut&Paste.

**Reply by LWinterberg** [reply]
Fwiw, Cut + paste generates clips on it's own, but I can see why you don't think you need those drag bars. I've got one more question though: In what way are the drag bars annoying? Like, do you keep accidentally clicking them when trying to select stuff? Do they prevent you from seeing the things that matter? 

I'm asking because, while the complaints you have with it are valid, I'm wondering if there's a better option for it design-wise than hiding it away in some preference menu that few people will find.

**Reply by petersampsonaudacity** [reply]
>I'm wondering if there's a better option for it design-wise than hiding it away in some preference menu that few people will find.
I am specifically not asking for it to be hidden away - that is precisely why I said that I thought it should be "on" by default.

What I **_am_** asking for is the ability to be **_able_** to hide the drag-bars for those of us that do not need or want them, with the ability to turn them back on again if and when we ever do need them.

For such users this:
a) frees up valuable screen real estate
b) removes the possibility of inadvertently moving a clip or track

In terms of "hiding it away" that is precisely why I suggested a command  - possibly in addition to a linked preference - _Audacity already does this for some other settings_.  I would recommend adding such a command to the Tracks menu, or possibly the View menu.

**Reply by petersampsonaudacity** [reply]
> Fwiw, Cut + paste generates clips on it's own
Yes, but only if you paste into an empty region of a track - if you paste into an existing track or clip it gets incorporated into the existing track or clip.

**Comment by Tantacrul** [comment]
There is always a concern whenever a new change is made that an option should be put in place to enable / disable it or to provide a method of toggling to the 'old way'. My feeling is that the drag bars shouldn't obstruct regular usage of Audacity, so I'm (presently) reluctant to add an option like this. After the user testing I've done, I'm reasonably confident that this change won't cause too much disruption - although some users will almost certainly look for the TimeShift tool and wonder where it has gone before noticing the clip handles. We've got a little work to do to make it clear that this change has been made when a user first opens Audacity.

**Reply by SteveDaulton** [reply]
I'm finding it noticeably less easy to work with "collapsed" tracks in large multi-track projects.
Annoyingly, it's very easy to accidentally move a clip when trying to make an audio selection.

(I'm guessing that there has not yet been much user testing with big multi-track projects, but with proposed changes such as real-time ability, this is likely to become a more common use case.)
